ngraph_bridge/ngraph_executor.cc

```diff
@@ -240,190 +220,167 @@ Status NGraphExecutor::GetNgExecutable(
+NGraphExecutor::GetNgExecutable(std::string signature,
+                                std::shared_ptr<ngraph::Function>& ng_function,
+                                const string serialized_ng_func,
+                                ng::runtime::Backend*& op_backend) {
+  std::shared_ptr<ngraph::runtime::Executable> ng_exec;
+
+  ngraph::Event event_compile("Compile nGraph", m_node_name, "");
+  BackendManager::LockBackend(m_op_backend_name);
   try {
-    ng_function = ngraph::deserialize(it->second);
+    if (m_do_aot) {
+      auto itr = m_aot_execs.find(signature);
+      if (itr == m_aot_execs.end()) {
+        BackendManager::UnlockBackend(m_op_backend_name);
+        return std::make_pair(
+            errors::Internal(
+                "Requested AOT, but could not find string with the "
+                "signature: ",
+                signature),
+            nullptr);
+      }
+      stringstream serialized_exec_read;
+      serialized_exec_read << (itr->second);
+      ng_exec = op_backend->load(serialized_exec_read);
+    } else {
+      ng_exec = op_backend->compile(ng_function);
+    }
   } catch (const std::exception& exp) {
-    return errors::Internal("Failed to deserialize ngraph function");
+    BackendManager::UnlockBackend(m_op_backend_name);
+    Status st = StringToFile("tf_function_error_" + m_node_name + ".json",
+                             serialized_ng_func);
+    string status_string =
+        "Caught exception while compiling op_backend: " + string(exp.what()) +
+        (st.ok() ? "" : (" Also error in dumping serialized function: " +
+                         st.error_message()));
+    return std::make_pair(errors::Internal(status_string), nullptr);
+  } catch (...) {
+    BackendManager::UnlockBackend(m_op_backend_name);
+    Status st = StringToFile("tf_function_error_" + m_node_name + ".json",
+                             serialized_ng_func);
+    string status_string =
+        "Error in compiling op_backend." +
+        (st.ok() ? "" : (" Also error in dumping serialized function: " +
+                         st.error_message()));
+    return std::make_pair(errors::Internal(status_string), nullptr);
   }
-  return Status::OK();
+  BackendManager::UnlockBackend(m_op_backend_name);
+  event_compile.Stop();
+  ngraph::Event::write_trace(event_compile);
+
+  return std::make_pair(Status::OK(), ng_exec);
+}
+
+//---------------------------------------------------------------------------
+//  NGraphExecutor::DestroyCallback
+//---------------------------------------------------------------------------
+void NGraphExecutor::DestroyCallback(
+    std::tuple<std::shared_ptr<ngraph::runtime::Executable>, std::string,
+               shared_ptr<PipelinedTensorsStore>>
+        evicted_ng_item,
+    ng::runtime::Backend*& op_backend) {
+  std::shared_ptr<ngraph::runtime::Executable> evicted_ng_exec;
+  std::tie(evicted_ng_exec, std::ignore, std::ignore) = evicted_ng_item;
```

Comment:
i see we are ignoring the other two. String might be fine. But pipelined_tensor_store might need a reset

Reply:
I understand what you are saying, but I don't think we need to reset pipelined_tensor_store here, at it get erased in the cache.